data-model-for-portal/Microsoft.DataFactory/datafactory.json

```diff
@@ -0,0 +1,125 @@
+{
+    "resourceType": "datafactory",
+    "apiVersion": "2018-06-01",
+    "learnMore": {
+        "url": "https://docs.microsoft.com/cli/azure/datafactory"
+    },
+    "commands": [
+        {
+            "name": "az datafactory list",
+            "description": "Lists factories. And Lists factories under the specified subscription.",
+            "path": "/subscriptions/{subscriptionId}/resourceGroups/{resourceGroupName}/providers/Microsoft.DataFactory/factories",
+            "confirmation": false,
+            "help": {
+                "learnMore": {
+                    "url": "https://docs.microsoft.com/cli/azure/datafactory#az-datafactory-list"
+                },
+                "parameterSets": [
+                    {
+                        "parameters": [
+                            "[--resource-group]",
+                            "[--subscription]"
+                        ]
+                    }
+                ]
+            },
+            "examples": [
+                {
+                    "description": "List all the data factories under a certain resource group.",
+                    "parameters": [
+                        {
+                            "name": "--subscription",
+                            "value": "[path.subscriptionId]"
+                        },
+                        {
+                            "name": "--resource-group",
+                            "value": "[path.resourceGroupName]"
+                        }
+                    ]
+                }
+            ]
+        },
+        {
+            "name": "az datafactory show",
+            "description": "Gets a factory.",
+            "path": "/subscriptions/{subscriptionId}/resourceGroups/{resourceGroupName}/providers/Microsoft.DataFactory/factories/{factoryName}",
+            "confirmation": false,
+            "help": {
+                "learnMore": {
+                    "url": "https://docs.microsoft.com/cli/azure/datafactory#az-datafactory-show"
+                },
+                "parameterSets": [
+                    {
+                        "parameters": [
+                            "[--factory-name]",
```

Comment:
Use `--name` to keep consistent with example below.
Same for `az datafactory delete`